core/solver/bicgstab.hpp

```diff
@@ -127,8 +132,14 @@ class Bicgstab : public EnableLinOp<Bicgstab<ValueType>> {
             preconditioner_ = matrix::Identity<ValueType>::create(
                 this->get_executor(), this->get_size()[0]);
         }
-        if (parameters_.criterion) {
-            stop_criterion_factory_ = std::move(parameters_.criterion);
+        if (parameters_.criteria.size() == 1) {
+            stop_criterion_factory_ = std::move(parameters_.criteria[0]);
+        } else if (parameters_.criteria.size() > 1) {
+            auto exec = parameters_.criteria[0]->get_executor();
+            stop_criterion_factory_ =
+                stop::Combined::build()
+                    .with_criteria(std::move(parameters_.criteria))
+                    .on(exec);
```

Comment:
Is it possible to make a function out of this, as this code is getting relatively long and is required for every solver?

Reply:
Made a function template that encapsulates this part of code logic.